src/cpp/src/speculative_decoding/continuous_batching_for_speculative_decoding_impl.cpp

```diff
@@ -197,7 +194,6 @@ UpdateRequestResult
 ContinuousBatchingPipeline::ContinuousBatchingForSpeculativeDecodingImpl::init_request_by_candidate(
     uint64_t request_id,
     const GeneratedSequences& candidates) {
-    _pull_awaiting_requests();
```

Comment:
we need to stay this line here because in `draft_model` requests will be running, and in `main_model` they be awaiting... in init
so, my suggestion is:
```
    if (m_requests.empty()) {
        _pull_awaiting_requests();
    }
```